src/dev-app/autocomplete/BUILD.bazel

```diff
@@ -0,0 +1,27 @@
+package(default_visibility = ["//visibility:public"])
+
+load("@io_bazel_rules_sass//:defs.bzl", "sass_binary")
+load("//tools:defaults.bzl", "ng_module")
+
+ng_module(
+    name = "autocomplete",
+    srcs = glob(["**/*.ts"]),
+    assets = [
+        "autocomplete-demo.html",
+        ":autocomplete_demo_scss",
+    ],
+    deps = [
```

Comment:
Should all of these ng_modules for each of the components dev app modules include deps on `@npm//@angular/{core,common}`  since they are creating `NgModule` and use `CommonModule`?


It seems like these are transitive dependencies?  I don't know if we have logic in place to prevent transitive dependencies.

Reply:
`@angular/core` is always brought in automatically through the macro. I tried to make dependencies as strict as possible but there might be cases where I missed some.

I think it's worth revisiting this once the `ng_module` has strict dependency enforcement implemented. Unlike `ts_library` this is not supported yet.. so there is no way to validate/enforce this. With Ivy it should be possible to use a `ts_library` with plugin.

Reply:
Oh interesting, I didn't think about how we had the defaults.

Yeah, I would really love to get to a point of using ts_library for everything.

Reply:
Yeah, we did that because `ng_module` doesn't work if `@angular/core` is not a dep, so it's always required and could be part of the macro defaults.

Yeah, using `ts_library` would be awesome. Strict deps are a big benefit.